commands.js

```diff
@@ -196,22 +196,17 @@ exports.commands = {
 				if (!targetRoom || targetRoom === Rooms.global) return this.errorReply('The room "' + innerTarget + '" does not exist.');
 				if (targetRoom.staffRoom && !targetUser.isStaff) return this.errorReply('User "' + this.targetUsername + '" requires global auth to join room "' + targetRoom.id + '".');
 				if (targetRoom.modjoin) {
-					if (targetRoom.auth && (targetRoom.isPrivate === true || targetUser.group === ' ') && !(targetUser.userid in targetRoom.auth)) {
+					if (targetRoom.auth && (targetRoom.isPrivate === true || targetUser.group === ' ')) {
+						if (!(user.userid in targetRoom.auth)) {
+							return this.errorReply('The room "' + innerTarget + '" does not exist.');
+						}
 						this.parse('/roomvoice ' + targetUser.name, false, targetRoom);
-						if (!(targetUser.userid in targetRoom.auth)) {
-							return;
+						if (Config.groupsranking.indexOf(targetRoom.auth[targetUser.userid] || ' ') < Config.groupsranking.indexOf(targetRoom.modjoin) && !targetUser.can('bypassall')) {
```

Comment:
Its the same check that's currently being done, just in a different place in the code...